cloudinit/net/network_state.py

```diff
@@ -800,26 +803,14 @@ def handle_wifis(self, command):
     def _v2_common(self, cfg) -> None:
         LOG.debug("v2_common: handling config:\n%s", cfg)
         for iface, dev_cfg in cfg.items():
-            if "set-name" in dev_cfg:
-                set_name_iface = dev_cfg.get("set-name")
-                if set_name_iface:
-                    iface = set_name_iface
             if "nameservers" in dev_cfg:
                 search = dev_cfg.get("nameservers").get("search", [])
                 dns = dev_cfg.get("nameservers").get("addresses", [])
                 name_cmd = {"type": "nameserver"}
                 if len(search) > 0:
-                    name_cmd.update({"search": search})
+                    name_cmd["search"] = search
```

Comment:
We could similarly exploit the falsyness of `search` and `dns` (no need to check that `len() > 0`) and also remove the default values in `.get("search", [])` and `.get("addresses", [])`, if you want.